internal/component/local/file_match/watch.go

```diff
@@ -48,6 +50,11 @@ func (w *watch) getPaths() ([]discovery.Target, error) {
 			}
 			continue
 		}
+
+		if w.ignoreOlder != 0 && fi.ModTime().Before(time.Now().Add(-w.ignoreOlder)) {
+			continue
+		}
```

Comment:
@wildum I was intentionally adding that before the dir check to prevent from navigating inside that directory if the files are not updated. Do you think we should change that ?

Reply:
the fi.IsDir() is just a boolean operation to check whether it's a file or a dir (see https://github.com/golang/go/blob/master/src/io/fs/fs.go#L231). I think that it makes sense to put it before and avoid system calls from the time.Now() when it's not needed

Reply:
let's move it after fi.IsDir() then?

Reply:
If we move it after then we would be unnecessarily navigating to all the files in that directory ? Do you think we should still move it after the check ?

Reply:
Directory modification time isn't propagated up the tree if a file is updated/added/etc multiple subdirectories down. IE if you have `test/log/my.log` and `my.log` is updated, while the `log` directory's modification time will change in most OSes the `test` directory's modification time will not. You shouldn't trust directory modification time for this purpose.

Reply:
Ohh interesting, Thats insightful, I have made the changes Thanks